### PR TITLE
rgw: use rados hints

### PR DESCRIPTION
**Pull request by yehudasa**

### CODE REVIEW

File: src/rgw/rgw_rados.cc [hunk 1]
```diff
@@ -6634,6 +6630,16 @@ int RGWRados::aio_put_obj_data(void *ctx, rgw_obj& obj, bufferlist& bl,
...
 
   if (ofs == -1) {
+    /*
+     * not setting size for allocation hint
+     * assuming first write buffer length is the common chunk size that will be used.
+     * Not using configured chunk size, because that might be modified when pool
+     * alignment requires.
+     */
+    ldout(cct, 20) << "setting rados allocation hint, write_size=" << bl.length() << dendl;
+    op.set_alloc_hint2(-1, bl.length(), ALLOC_HINT_FLAG_SEQUENTIAL_WRITE |
```
Comment (liewegas) [inline src/rgw/rgw_rados.cc]: 0, not -1, for no hint

File: src/rgw/rgw_rados.cc [hunk 2]
```diff
@@ -6576,14 +6584,14 @@ int RGWRados::put_system_obj_data(void *ctx, rgw_obj& obj, bufferlist& bl,
...
-
-  if (ofs == -1) {
-    op.write_full(bl);
-  } else {
-    op.write(ofs, bl);
   }
+
+  op.set_alloc_hint2(bl.length(), bl.length(), ALLOC_HINT_FLAG_SEQUENTIAL_WRITE |
+                                               ALLOC_HINT_FLAG_SEQUENTIAL_READ |
+                                               ALLOC_HINT_FLAG_APPEND_ONLY);
```
Comment (liewegas) [inline src/rgw/rgw_rados.cc]: could put this in the exclusive block so that ti's only set on create

Comment (liewegas) [inline src/rgw/rgw_rados.cc]: i'd only use bl.length() here if it's actually expected to mean something...

**Comment by liewegas** [comment]
needs rebase

File: src/rgw/rgw_rados.cc (hunk 2 above)
Comment (yehudasa) [inline src/rgw/rgw_rados.cc]: @liewegas the exclusive block doesn't signify whether it's create or not. Object would be completely overwritten every time. Do we need to specify a hint here?

Comment (liewegas) [inline src/rgw/rgw_rados.cc]: It's harmless on an existing object; ideally it should only be called when the object is first created though (just for efficiency).

**Comment by tchaikov** [comment]
@yehudasa  ping?

**Comment by yehudasa** [comment]
This went under the radar for a while. Will look at it after my vacation, unless @oritwas has time for it.

**Comment by yehudasa** [comment]
@liewegas @tchaikov @mattbenjamin rebased

**Comment by mattbenjamin** [comment]
@yehudasa should we revive this?  sounds like it

**Comment by yehudasa** [comment]
reopening

**Comment by mattbenjamin** [comment]
I'll update this PR in the Pacific cycle--we're testing a downstream version of the first commit now.

Matt

**Pull request closed without merging**